### PR TITLE
Demo aug 19

### PR DESCRIPTION
**Pull request by PHSAServiceOperationsCenter**
check rules and apppoc

### CODE REVIEW

File: pocApp/new_tasks.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+@shared_task
+def go():
+    return True
+    for i in range(10):
+        print ("I am @ record:%d " % (i))
+
+@task()
+def getnmapdata():
+    rc = "success"
+    nodeList = OrionSslNode.ip_addresses()
```
Comment (serbant) [inline pocApp/new_tasks.py]: node_list (not nodeList) must be populated using the nodes method, not the ip_addresses method. it must include both orionnode_orionid and orionapmapplication_orionid

File: pocApp/new_tasks.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    #nm = NmapProcess(param, options='--script ssl-cert')
+    for node in nodeList:
+        nm = NmapProcess(node, options='--script ssl-cert')
```
Comment (serbant) [inline pocApp/new_tasks.py]: the NmapProcess calls need to execute each in its celery process-thread combination and need to be segregated by port

Comment (serbant) [inline pocApp/new_tasks.py]: you must catch nmap execution errors and you need to handle the case when nmap returns a filtered (firewalled) host, post tuple

File: pocApp/new_tasks.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        nm.run()
+        xmlData = nm.stdout
+        doc = xml.dom.minidom.parseString(xmlData )
+        #doc = xml.dom.minidom.parse(xmlData )
+        output = []
+        for host in doc.getElementsByTagName("host"):
```
Comment (serbant) [inline pocApp/new_tasks.py]: xml processing needs to happen into a separate function or method

File: pocApp/new_tasks.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                    #except:
+                    #    pass
+            
+                #output.append(ip + ',' + commonName + ',' + organizationName + ',' + countryName + ',' + 
+                #notBefore + ',' + notAfter)
+                output.append(commonName + ":" + notBefore + ',' + notAfter + ", " + host_name + ", " +  md5 + ',' + sha1)
+                logger.info("output: ", output)
+                print("Hello:-----------------------",output)
+                try:
+                    dbCerts=MyCertsData()
```
Comment (serbant) [inline pocApp/new_tasks.py]: we need to come up with a better way of handling the data. please hang a story on the backlog about ssl_certs history

File: pocApp/serializers.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# api/serializers.py
```
Comment (serbant) [inline pocApp/serializers.py]: why do we need this?

File: pocApp/utils/db_helper.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
```
Comment (serbant) [inline pocApp/utils/db_helper.py]: replace this with django facilities, please. or remove it altogether

File: pocApp/urls.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+from django.conf.urls import url
+#from rest_framework.urlpatterns import format_suffix_patterns
+from  .import views
+
+urlpatterns = [
+    url(r'^$', views.index, name="index"),
+    url(r'getnmapdata/$',  views.nMapData,  name="nMapData"),
```
Comment (serbant) [inline pocApp/urls.py]: do these views exist or are they being used anywhere?

File: pocApp/utils/nmap_get_xml.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
```
Comment (serbant) [inline pocApp/utils/nmap_get_xml.py]: is this being used anywhere? if so, please justify

File: pocApp/utils/nmap_parse_xml.py [hunk 9]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
```
Comment (serbant) [inline pocApp/utils/nmap_parse_xml.py]: same as previous comment

File: pocApp/views.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
+from django.shortcuts import render
```
Comment (serbant) [inline pocApp/views.py]: is this being of any use? if not, please remove

File: rules/apps.py [hunk 11]
```diff
@@ -0,0 +1,4 @@
+from django.apps import AppConfig
+
+class RulesConfig(AppConfig):
```
Comment (serbant) [inline rules/apps.py]: see comments about verbose_name and gettext_lazy

File: rules/models.py [hunk 12]
```diff
@@ -0,0 +1,16 @@
+from __future__ import unicode_literals
```
Comment (serbant) [inline rules/models.py]: imports from __future__ are only relevant for Python 2. we are using Python 3; please remove all from __future__ import statements

Comment (serbant) [inline rules/models.py]: please hang a story on the backlog about this item and add a design task to said story. this needs to apply to everything, not just certs.
the structure should be rule_name, rule_applies_to(foreign key), rule_type(enum: exists, interval, missing), etc, etyc

File: rules/new_tasks.py [hunk 13]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
```
Comment (serbant) [inline rules/new_tasks.py]: rename this to tasks

Comment (serbant) [inline rules/new_tasks.py]: this needs to be seriously abstracted. there has to be a module or class where a rule can be initialized based on the data in the rules model. rule execution needs to be celerified or at least developed with celery in mind

File: configs/rpms/os_rpm.info [hunk 14]
```diff
@@ -0,0 +1,701 @@
+redhat-lsb-submod-multimedia-4.1-27.el7.centos.1.x86_64
```
[the remark this answers is not in the document]
Comment (PHSAServiceOperationsCenter) [reply configs/rpms/os_rpm.info]: Done

File: pocApp/junk/a1.py.bak [hunk 15]
```diff
@@ -0,0 +1,18 @@
+from celery.task.schedules import crontab
```
[the remark this answers is not in the document]
Comment (PHSAServiceOperationsCenter) [reply pocApp/junk/a1.py.bak]: done

**Pull request closed without merging**